src/lib.rs

```diff
@@ -132,6 +132,9 @@
 
 #![deny(rust_2018_idioms)]
 #![deny(rustdoc::broken_intra_doc_links)]
+#![deny(clippy::all)]
+#![cfg_attr(feature = "cargo-clippy", deny(warnings))]
```

Comment:
How does this work?

Reply:
`clippy` enables the `cargo-clippy` feature on all crates when running through it, automatically denying all `rustc` warnings in this instance. However, this implies warnings from `clippy::all` (that lint group should only include warn/deny lints) and the `#![deny(clippy::all)]` is thus superfluous.

Reply:
deny clippy all means that it'll error on clippy warnings instead of treating them as warnings.

Reply:
But that's also what `#![cfg_attr(feature = "cargo-clippy", deny(warnings))]` does :)

Reply:
I think those about rustc warnings, and not clippy onces? I think it was like that in the past, at least.